docs/device_support_guide.md

```diff
@@ -0,0 +1,184 @@
+This **Value** will change depending on the parameters related to it.
+
+Some parameters will be changing automatically without your intervention (e.g. the CPU temperature), others will change in response to different settings in the MSI app (e.g. battery charge limits)
+
+Example: To figure out which **Addresses** are used by user scenarios (or shift
+modes) go to the user scenario page in the MSI app and keep switching the scenarios
+while looking at the EC table.
+
+Eventually you'll notice one or more values that change each time you change the
+setting, once you find them, you can start writing down the addresses and values
+corresponding to *each* user scenario, so you can report them.
+
+# Linux method
+
+This method is very limited compared to the Windows method, because most
+MSI laptop features are tied to software toggles that can only be found in the official Windows apps.
+
+If you are lucky, your laptop model will have similar addresses to another laptop
+that is already supported by the driver, but usually this only happens on some
+features like cooler boost and battery charge limit.
+
+On Linux, you can do the EC dump in several ways.
++ `msi-ec` built-in debug mode (git version only)
++ `ec-sys` kernel module
++ By reading the memory that is virtually mapped to EC
+
+To request support for your device model, make the EC dump in text form
+and open the [Issue on Github](https://github.com/BeardOverflow/msi-ec/issues/new?assignees=&labels=New+firmware&projects=&template=support_request.yml).
+
+## `MSI-EC` debug mode
+
+Make sure that you are running the git version of the module.
+If not, follow the installation guide in the [Readme](../README.md#Installation) file.
+
+Unload module if it was previously loaded:
+
++ `sudo modprobe -r msi-ec`
+
+Load module with debug mode enabled:
+
++ `sudo modprobe msi-ec debug=true`
+
+Make a dump by reading the module pseudo file:
+
++ `cat /sys/devices/platform/msi-ec/debug/ec_dump`
+
+Or save it to file:
+
++ `cat /sys/devices/platform/msi-ec/debug/ec_dump > ec_dump.txt`
+
+If you got the next error, then your module is not loaded in debug mode or doesn't have it:
+
+> `cat: /sys/devices/platform/msi-ec/debug/ec_dump: No such file or directory`
+
+## `EC-SYS` method
+
+> [!NOTE]
+> The `ec-sys` module may not be included in some distros, such as Fedora.
+
+To start, You need to load a module called `ec_sys`:
+
+* `sudo modprobe ec_sys`
+
+[//]: # (If you need to write to a specific address &#40;but you really shouldn’t&#41; you can enable)
+[//]: # (Read/Write mode for this module:)
+[//]: # ()
+[//]: # (* `sudo modprobe ec_sys write_support=1`)
+
+After that you can extract the EC table and print it on the terminal in text form:
+
+* `hexdump -C /sys/kernel/debug/ec/ec0/io`
+
+Or save it to file:
+
++ `hexdump -C /sys/kernel/debug/ec/ec0/io > ec_dump.txt`
+
+If you need dump in binary form:
+
+* `cat /sys/kernel/debug/ec/ec0/io > ec_dump.bin`
```

Comment:
I don't think anyone does. At least, we don't need binaries submitted here, so let's exclude this part